crypto/evp/mac_lib.c

```diff
@@ -189,3 +189,65 @@ int EVP_MAC_names_do_all(const EVP_MAC *mac,
 
     return 1;
 }
+
+unsigned char *EVP_Q_mac(OSSL_LIB_CTX *libctx, const char *name, const char *propq,
+                         const char *subalg, const OSSL_PARAM *params,
+                         const void *key, int keylen,
```

Comment:
keylen should be a size_t I think

Reply:
Changing this would break the compatibility decl
```
#   define HMAC(evp_md, key, keylen, data, datalen, out, outlen) \
    EVP_Q_mac(NULL, "HMAC", NULL, EVP_MD_name(evp_md), NULL, \
              key, keylen, data, datalen, out, EVP_MD_size(evp_md), outlen)
```

Reply:
You could cast here. But of course not for the output parameter.

Reply:
Ah, right! Or rely on an implicit conversion from `int` to `size_t`.
Which is the preferred way?

Reply:
Lets make it explicit

Reply:
Done